public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx

```diff
@@ -180,38 +170,28 @@ export class ConfigEditor extends PureComponent<Props, State> {
       azureMonitorUrl +
       `/${subscriptionId}/providers/Microsoft.OperationalInsights/workspaces?api-version=2017-04-26-preview`;
 
-    return this.backendSrv
-      .datasourceRequest({
-        url: this.state.config.url + workspaceListUrl,
-        method: 'GET',
-      })
-      .then((result: any) => {
-        return result.data.value.map((val: any) => {
-          return {
-            value: val.properties.customerId,
-            label: val.name,
-          };
-        });
-      })
-      .catch((error: any) => {
-        throw error;
-      });
+    const result = await this.backendSrv.datasourceRequest({
+      url: this.props.options.url + workspaceListUrl,
+      method: 'GET',
+    });
+
+    return ResponseParser.parseWorkspacesForSelect(result);
   };
 
   getSubscriptions = async () => {
     if (!this.hasNecessaryCredentials()) {
       return;
     }
 
-    const subscriptions = (await this.loadSubscriptions()) || [];
+    const subscriptions = ((await this.loadSubscriptions()) || []) as SelectableValue[];
 
     if (subscriptions && subscriptions.length > 0) {
       this.setState({ subscriptions });
 
-      this.state.config.jsonData.subscriptionId = this.state.config.jsonData.subscriptionId || subscriptions[0].value;
+      this.props.options.jsonData.subscriptionId = this.props.options.jsonData.subscriptionId || subscriptions[0].value;
```

Comment:
Looks weird that we're mutating a prop on `this.props` here.

Reply:
Oh yeah- this was something I was doing on the state. I can set these on the state.